packages/p2p/src/components/my-profile/block-user/block-user.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { observer } from 'mobx-react-lite';
+import { useStores } from 'Stores';
+import { DesktopWrapper, MobileFullPageModal, MobileWrapper } from '@deriv/components';
+import BlockUserModal from 'Components/block-user/block-user-modal';
+import BlockUserTable from 'Components/my-profile/block-user/block-user-table/block-user-table';
+import SearchBox from 'Components/search-box';
+import { my_profile_tabs } from 'Constants/my-profile-tabs';
+import debounce from 'lodash.debounce';
+import { localize } from 'Components/i18next';
+
+const BlockUser = () => {
+    const { general_store, my_profile_store } = useStores();
+
+    const loadBlockedAdvertisers = debounce(search => {
+        my_profile_store.setSearchTerm(search.trim());
+        my_profile_store.loadMoreBlockedAdvertisers();
+    }, 200);
+
+    const onSearch = search => {
+        // Ensures that blocked advertisers list is not reloaded if search term entered is the same
+        if (my_profile_store.search_term !== search.trim()) {
```

Comment:
There is a bug where if the user presses non-alpahetical/numerical keys such as Shift, the search box will run `onSearch`, thus this is to prevent flickering/re-rendering of the list of blocked users if the user has the same search term as before